docs/Reference/CLI/CLI-Subcommands.md

```diff
@@ -59,3 +59,314 @@ Generate a list of peer IDs including the private key, public key, and peer ID w
+```bash tab="Example"
+teku validator generate --deposit-amount-gwei=32000000000
+```
+
+Amount to deposit in the Ethereum 1.0 deposit contract.
+
+#### encrypted-keystore-enabled
+
+```bash tab="Syntax"
+teku validator generate --encrypted-keystore-enabled=<BOOLEAN>
+```
+
+```bash tab="Example"
+teku validator generate --encrypted-keystore-enabled=false
+```
+
+Specify whether to create encrypted V3 keystore validator and withdrawal keys.
+Defaults to `true`.
+
+Use [`teku validator generate --keys-output-path`](#keys-output-path) to specify the
+location of the encrypted or unencrypted keys.
+
+#### eth1-deposit-contract-address
+
+```bash tab="Syntax"
+teku validator generate --eth1-deposit-contract-address=<ADDRESS>
+```
+
+```bash tab="Command Line"
+teku validator generate --eth1-deposit-contract-address=0x77f7bED277449F51505a4C54550B074030d989bC
+```
+
+Ethereum 1.0 address of deposit contract.
+
+#### eth1-endpoint
+
+```bash tab="Syntax"
+teku validator generate --eth1-endpoint=<URL>
+```
+
+```bash tab="Command Line"
+teku validator generate --eth1-endpoint=http://localhost:8545
+```
+
+JSON-RPC URL of Ethereum 1.0 node.
+
+#### eth1-keystore-file
+
+```bash tab="Syntax"
+teku validator generate --eth1-keystore-file=<FILE>
+```
+
+```bash tab="Command Line"
+teku validator generate --eth1-keystore-file=/home/me/me_node/keystore
+```
+
+Path to the encrypted V3 keystore file containing the Ethereum 1.0 private key used to
+send the deposit transaction.
+
+Cannot be used with `teku validator generate --eth1-private-key`
+
+#### eth1-keystore-password-file
+
+```bash tab="Syntax"
+teku validator generate --eth1-keystore-password-file=<FILE>
+```
+
+```bash tab="Command Line"
+teku validator generate --eth1-keystore-password-file=/home/me/me_node/password
+```
+
+Path to the file containing the password to decrypt the V3 keystore.
+
+#### eth1-private-key
+
+```bash tab="Syntax"
+teku validator generate --eth1-private-key=<KEY>
+```
+
+```bash tab="Command Line"
+teku validator generate --eth1-private-key=8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63
+```
+
+Ethereum 1.0 private key to send the deposit transaction. Cannot be used with
+`teku validator generate --eth1-keystore-file`.
+
+#### keys-output-path
+
+```bash tab="Syntax"
+teku validator generate --keys-output-path=<FILE|DIR>
+```
+
+```bash tab="Command Line"
+teku validator generate --keys-output-path=/home/me/me_node/keys
+```
+
+Specify the output location for validator and withdrawal keys.
+
+* Path to the output file when using unencrypted keys. That is, [`teku validator generate --encrypted-keystore-enabled`](#encrypted-keystore-enabled)
+set to `false`.
+
+* Path to the output directory for the auto-generated V3 keystore files. That is,
+[`teku validator generate --encrypted-keystore-enabled`](#encrypted-keystore-enabled) set to `true`.
+
+If not set, unencrypted keys are written to standard out, and encrypted keystores are created in current directory.
```

Comment:
If set to `false' then we write unencrpted files. By default, it is set to true.